libcaf_core/caf/scheduled_actor.hpp

```diff
@@ -354,6 +361,31 @@ class CAF_CORE_EXPORT scheduled_actor : public abstract_scheduled_actor,
   }
 #endif // CAF_ENABLE_EXCEPTIONS
 
+  /// Sets a custom handler for timeouts that trigger after *not* receiving
+  /// a message for a certain amount of time.
+  template <class RefType, class RepeatType>
+  void set_idle_handler(timespan delay, RefType, RepeatType, idle_handler fun) {
+    if (delay == infinite)
+      return;
```

Comment:
Technically speaking, with a strong reference, shouldn't the behavior be that the actor never dies?

I don't think it's practical to do that, but maybe this should be an assertion instead when the reference is strong and the timeout infinite.

Reply:
You're right, we shouldn't silently discard it. It's an error now. 👍